pkg/agent/watcher.go

```diff
@@ -43,18 +44,40 @@ func NewWatcher(configDir string, modelDir string) Watcher {
 	if err != nil {
 		log.Error(err, "Failed to sync model dir")
 	}
-	return Watcher{
+	watcher := Watcher{
 		configDir:    configDir,
 		modelTracker: modelTracker,
-		ModelEvents:  make(chan ModelOp),
+		ModelEvents:  make(chan ModelOp, 100),
```

Comment:
I thought we intentionally made this a blocking channel? Can @ifilonenko can confirm?

Reply:
I do not see much reason to make it blocking channel, the problem is that if we push the events first and then start the puller the process just hangs there.